src/SerializableLimitIterator.php

```diff
@@ -79,8 +79,9 @@ public function unserialize($data)
 
     public function __unserialize(array $data)
     {
-        $this->__construct($data['it'], $data['offset'], $data['count']);
-        $this->seek($data['pos'] + $data['offset']);
+        $this->offset = (int) $data['offset'];
+        $this->count  = (int) $data['count'];
+        parent::__construct($data['it'], $this->offset, $this->count);
```

Comment:
Any way we can test how it was broken before? :thinking:

Reply:
Calling a constructor directly from another method is valid PHP code (though not the best practice) so it's not about testing how it was broken.
It's only Psalm that reports errors [here](https://github.com/laminas/laminas-paginator/actions/runs/4238083232/jobs/7364769580?pr=55#step:3:359) and [here](https://github.com/laminas/laminas-paginator/actions/runs/4228089138/jobs/7343193290?pr=55#step:3:354)

Reply:
Besides the constructor and Psalm: the call of `seek` has been removed without any effect, which indicates that tests are missing.

Reply:
Oops, missing _seek_ issue is fixed now